sympy/integrals/integrals.py

```diff
@@ -439,6 +439,34 @@ def doit(self, **hints):
         # There is no trivial answer and special handling
         # is done so continue
 
+        # first make sure any definite limits have integration
+        # variables with matching assumptions
+        reps = {}
+        for xab in self.limits:
+            if len(xab) != 3:
+                continue
+            x, a, b = xab
+            l = (a, b)
+            if all(i.is_nonnegative for i in l) and not x.is_nonnegative:
+                d = Dummy(positive=True)
+            elif all(i.is_nonpositive for i in l) and not x.is_nonpositive:
+                d = Dummy(negative=True)
+            elif all(i.is_real for i in l) and not x.is_real:
```

Comment:
Is it a problem having extendedreal here? I guess for integrals with oo bounds it would help.

Reply:
I was going off of @asmeurer's comment that the value of an integral at the limit will not matter...so I am just keeping it simple. If you can find a case where it matters we could make the change.

Reply:
real is more restricted than extended real, so it should lead to more simplifications. Unless we can get a wrong answer from having real here instead of extended real I would just stick with real.

Reply:
> integrals with oo bounds

Such (improper) integrals are limits of integrals taken over finite intervals. The integration variable should not take infinite values.

Reply:
I was primarily considering the type of integral as in the in this PR edited Wester test. Now, it wasn't the y-variable that caused the major part of the simplification, but I guess that there may be cases where this would be the case.

But as you say, easy to change if there are cases where it actually matters.

Reply:
You are off course correct @jksoum, still:
```
$ git grep Integral | grep oo | wc -l
145
```

Reply:
Forgot this:
```
$ git grep integrate | grep oo | wc -l
330
```